src/database/GameLoader.ts

```diff
@@ -33,54 +35,46 @@ export class GameLoader implements IGameLoader {
     return GameLoader.instance;
   }
 
-  public async add(game: Game): Promise<void> {
-    const d = await this.idsContainer.getGames();
-    d.games.set(game.id, game);
-    if (game.spectatorId !== undefined) {
-      d.participantIds.set(game.spectatorId, game.id);
-    }
-    for (const player of game.getPlayers()) {
-      d.participantIds.set(player.id, game.id);
-    }
+  public add(game: Game): void {
+    return this.cache.add(game);
   }
 
   public async getLoadedGameIds(): Promise<Array<GameIdLedger>> {
-    const d = await this.idsContainer.getGames();
+    const d = await this.cache.get();
     const map = new MultiMap<GameId, SpectatorId | PlayerId>();
     d.participantIds.forEach((gameId, participantId) => map.set(gameId, participantId));
     const arry: Array<[GameId, Array<PlayerId | SpectatorId>]> = Array.from(map.associations());
     return arry.map(([id, participants]) => ({id: id, participants: participants}));
   }
 
   public async getByGameId(gameId: GameId, bypassCache: boolean): Promise<Game | undefined> {
-    const d = await this.idsContainer.getGames();
+    const d = await this.cache.get();
     if (bypassCache === false && d.games.get(gameId) !== undefined) {
       return d.games.get(gameId);
     } else if (d.games.has(gameId)) {
-      return this.loadGameAsync(gameId, bypassCache);
+      return this.loadGame(gameId, bypassCache);
     } else {
       return undefined;
     }
   }
 
   public async getByParticipantId(id: PlayerId | SpectatorId): Promise<Game | undefined> {
-    const d = await this.idsContainer.getGames();
+    const d = await this.cache.get();
+
     const gameId = d.participantIds.get(id);
-    if (gameId !== undefined && d.games.get(gameId) !== undefined) {
-      return d.games.get(gameId);
-    } else if (gameId !== undefined) {
-      return this.loadParticipant(id);
-    } else {
-      return undefined;
-    }
+    if (gameId === undefined) return undefined;
+
+    const cachedGame = d.games.get(gameId);
+    if (cachedGame !== undefined) return cachedGame;
+    return this.loadGame(gameId, true);
```

Comment:
What's happening here? If you read loadParticipant carefully, you'll see this is repeating a ton of what is already done, for no reason.